python/lib/tmdbscraper/tmdb.py

```diff
@@ -225,13 +226,39 @@ def _load_base_urls(url_settings):
             url_settings.setSetting('lastUpdated', str(_get_date_numeric(datetime.now())))
     return urls
 
-def _parse_trailer(trailers, fallback):
+def _parse_trailer(trailers, fallback, trailer_url_chk):
     if trailers.get('youtube'):
-        return 'plugin://plugin.video.youtube/?action=play_video&videoid='+trailers['youtube'][0]['source']
+        return _trailer_check(trailers.get('youtube'), trailer_url_chk)
     if fallback.get('youtube'):
-        return 'plugin://plugin.video.youtube/?action=play_video&videoid='+fallback['youtube'][0]['source']
+        return _trailer_check(fallback.get('youtube'), trailer_url_chk)
     return None
 
+def _trailer_check(youtube, trailer_url_chk):   
+    backup_sources = []
+    youtubeUrl = 'plugin://plugin.video.youtube/?action=play_video&videoid='
+    for option in youtube:
+        source = option.get('source')
+        if option.get('type') == 'Trailer':   
+            if not trailer_url_chk:           # trailer_url_chk OFF                             
+                return youtubeUrl + source    # video is defined as "Trailer" by TMDB.
+            elif _check_youtube(source):         # trailer_url_chk ON
+                return youtubeUrl + source       # video is available and is defined as "Trailer" by TMDB. Perfect link!
+            else:
+                backup_sources.append(source)    # Backing up sources in case it doesn't find any perfect link.
```

Comment:
Did you mean to put failed checks on the backup list? This implementation fetches each failed "Trailer" option twice.

Reply:
Looks like the last else ought to be de-dented once so that it pairs with the 'type' check rather than the youtube check.